### ISSUE TITLE
[BUG] mingw64 gfortran (10.2) - module name can't be longer than 6 characters

### ISSUE THREAD

**Issue by mehraneslaminia** [msys2/MINGW-packages#6908]
## Describe the bug

When a module name is less than 6 characters, the program builds with any issue. But, when I changed the module name to more than 6 characters, I'm getting the following error:

>f951.exe: Fatal Error: Reading module 'm123456.mod' at line 1 column 26: Unexpected EOF
compilation terminated.

## Steps to Reproduce the Problem

Fortran files:
```Fortran
! source1.f90
module M123456

integer :: A = 1

end module
```
```Fortran
! main.f90
program test

use M123456

write(*,*) "Hello World", A

end program
```

Compile command: `gfortran -o test.exe source1.f90 main.f90`

If I change the module name to M12345 (six characters) instead of M123456 (7 characters), the program builds correctly. 

> **Note**: gfortran on Ubuntu doesn't have any issue

**Comment by lazka** [comment]
@scivision do you know what could be the problem by any chance?

**Comment by scivision** [comment]
This usually means there is a .mod file in the directory from another compiler. I would try `rm *.mod` in that directory.

**Comment by scivision** [comment]
The Fortran 2003 [standard](https://wg5-fortran.org/N1551-N1600/N1579.pdf) set a 63 character name limit, including for modules.
Gfortran 10.2 and Intel oneAPI respect this limit. Increasing the module name length by one in this example will fail on both compilers:

```fortran
module a11111111111111111111111111111111111111111111111111111111111111
end module

program main
use a11111111111111111111111111111111111111111111111111111111111111
end program

```

**Comment by scivision** [comment]
tldr this is not an MSYS2 issue.

**Comment by mehraneslaminia** [comment]
The module is compiled with the same compiler. The compiler version is `GNU Fortran (Rev1, Built by MSYS2 project) 10.2.0`. 
The Fortran standard has a limit of 63 characters, but here we are talking about just **6** characters, not 63. 

I have tried sample codes with both `intel on windows` and `gfortran Ubuntu build`, there was no issue. I've even tried `GNU Fortran (x86_64-posix-seh-rev0, Built by MinGW-W64 project) 8.1.0`. This bug is specific with MSYS build of gfortran.

**Comment by scivision** [comment]
Yes I've also tried gfortran 10.2 via MSYS2, which I use everyday. I specifically tried your example. I would say try creating your example file in some new directory just to be sure there's no .mod file hanging around

**Comment by mehraneslaminia** [comment]
I just tried it. same issue. I even tried `m1234` (5 characters), it doesn't work either. For some reason, I can only work with modules with 6-character names.

**Comment by mehraneslaminia** [comment]
More info: when I tried to build my sample code directly from MSYS2-MinGW terminal there is no issue.

This issue only happens when I use Windows CMD or PowerShell. I have added mingw64\bin to my path.

**Comment by Liu-Jincan** [comment]
I had the same issue when I used msys-bash in the VScode. Running the code from MSYS2-MinGW terminal is a good choice to solve this problem. Thanks for your sharing.
